pinot-query-planner/src/main/java/org/apache/pinot/query/type/TypeSystem.java

```diff
@@ -50,4 +53,11 @@ public int getMaxNumericScale() {
   public int getMaxNumericPrecision() {
     return MAX_DECIMAL_PRECISION_DIGIT;
   }
+
+  @Override
+  public RelDataType deriveAvgAggType(RelDataTypeFactory typeFactory,
+      RelDataType argumentType) {
+    return typeFactory.createTypeWithNullability(
+        typeFactory.createSqlType(SqlTypeName.DOUBLE), false);
```

Comment:
This is not the right way to fix the issue. we should rely on Calcite's derived type system to figure out what the return type should be. this way all AGG results are returning as double which is going to cause more issue than it resolves.

Reply:
based on a coarse read, we should check the input decimal precision and try to do upcast to the appropriate floating point type. (e.g. the scale and precision

Reply:
Thanks for the review! I resorted to using double datatype for AVG because that's what we do in V1 engine. 

We could maybe do what postgres does. Thoughts? 
> numeric for any integer type argument, double precision for a floating-point argument, otherwise the same as the argument data type

Reply:
Based on offline discussion, created oss issue #10318 to determine the logic for AVG return type.